paper_framework_sources/p1898_forward_progress_delegation.md

```diff
@@ -274,103 +285,87 @@ vector<int> doWork(Sender<vector<int>> vec) {
-A `scheduler` representing that execution context will be made available from the `receiver` passed to `submit` on the `sender`, thus making that `receiver` a `scheduler_provider`.
+A `scheduler` representing that execution context will be made available from the `receiver` passed to `connect` on the `sender`, thus making that `receiver` a `scheduler_provider`.
 
-
-### Wording
+### Wording to modify final otherwise clause
  * Otherwise:
    * Constructs an execution context owned by the calling thread such that tasks enqueued into that context will be processed by the calling thread.
-   * Constructs a `scheduler`, `s` representing that execution context.
-   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::submit(S, r)`.
-   * `execution::get_scheduler(r)` will return `s`.   
-   * Waits on the synchronization primitive to block on completion of `S`.
+   * Constructs a `scheduler`, `sch` representing that execution context.
+   * Constructs a `receiver`, `r` over an implementation-defined synchronization primitive and passes that callback to `execution::connect(s, r)` such that:
+     * `r` satisfies `scheduler_provider`.
+     * `execution::get_scheduler(r)` will return `sch`.
+     * If `set_value` is called on `r`, the call to `sync_wait` will return the passed value.
+     * If `set_error` is called on `r`, the call to `sync_wait` will throw the passed value as an exception.
+     * If `set_done` is called on `r`, the call to `sync_wait` will throw a cancellation exception.
+   * Waits on the synchronization primitive to block on completion of `s`, driving the execution context as necessary until `set_value`, `set_error` or `set_done` is called on `r` .
 
 The thread that invokes `sync_wait` will block with forward progress delegation on completion of the work where supported by the implementation of `s`.
 
-
-## execution::on
+## execution::on modifications
 ### Summary
-Transitions execution from one executor to the context of a scheduler.
-Passes a `scheduler_provider` to the passed `sender`, such that it is valid to apply `on` to a sender that restricts its `submit` operation to `scheduler_provider`s.
-That is that:
-```
-sender1 | on(scheduler1) | bulk_execute(f)...
-```
+In addition to transitioning execution from one execution context to another, `on` also changes the `scheduler` that is propagated upstream through `scheduler_provider`s.
 
-will return a sender that runs in the context of `scheduler1` such that `f` will run on the context of `scheduler1`, potentially customized, but that is not triggered until the completion of `sender1`.
-If `sender1` calls `get_scheduler` on the `receiver` passed by `on` to `sender1`'s `submit` operation, then that shall return `scheduler1`.
+### Wording to modify otherwise clause
```

Comment:
From Kirk on P1897: The parameter to on() should be a scheduler_provider, not a scheduler.

Open discussion on this. Should schedule() work on a scheduler_provider? Should every scheduler be a scheduler_provider?

Reply:
It doesn't really require that calls call get_scheduler, it just requires that they construct a scheduler. The usual method would be to call get_scheduler on some context, or indeed on a receiver or whatever, but a context would be the normal case for the user of algorithms rather than the implementation. The POR in 443 is that the scheduler is the thing we pass around, so I think it makes sense to stick with that unless we have good reason to change.